### PR TITLE
lsp--capf-cached-items: dont group on prefix anymore

### PR DESCRIPTION
**Pull request by kiennq**
Since we're using `lsp--fuzzy-score`, there's no need for grouping candidates prefix anymore.

### CODE REVIEW

**Comment by yyoncho** [comment]
And there is one more thing that we should do - use `while-no-input` when processing the completions. When it is hit we should return empty/previous completion result because it will be recalculated either way.

**Comment by kiennq** [comment]
> And there is one more thing that we should do - use `while-no-input` when processing the completions. When it is hit we should return empty/previous completion result because it will be recalculated either way.

Good point, I will do that

File: lsp-mode.el [hunk 1]
```diff
@@ -4509,23 +4508,24 @@ We can pass LSP-ITEMS, which will be used when there's no cache.
...
+                                                     (setq fuz-queries
+                                                           (plist-put fuz-queries start-point s))
+                                                     s))))
+                               (when (string-match fuz-query cand)
+                                 (setq cand (copy-sequence cand))
+                                 (put-text-property 0 1 'match-data (match-data) cand)
+                                 (put-text-property 0 1 'completion-score (lsp--fuzzy-score query cand) cand)
+                                 cand)))
+                           items))
                   (-sort (-on #'> (lambda (o)
```
[the remark this answers is not in the document]
Comment (kiennq) [reply lsp-mode.el:4529]: Ok

[the remark this answers is not in the document]
Comment (kiennq) [reply lsp-mode.el:4529]: Sure

**Comment by yyoncho** [comment]
> > And there is one more thing that we should do - use `while-no-input` when processing the completions. When it is hit we should return empty/previous completion result because it will be recalculated either way.
> 
> Good point, I will do that

Hm, we should be careful, because this might not work, e. g. when user does company select next item we should not cancel the calculations...

File: lsp-mode.el [hunk 2]
```diff
@@ -4509,23 +4508,24 @@ We can pass LSP-ITEMS, which will be used when there's no cache.
...
+                                               (let ((s (buffer-substring-no-properties start-point (point))))
+                                                 (setq queries (plist-put queries start-point s))
+                                                 s)))
+                                    (fuz-query (or (plist-get fuz-queries start-point)
+                                                   (let ((s (lsp--regex-fuzzy query)))
+                                                     (setq fuz-queries
+                                                           (plist-put fuz-queries start-point s))
+                                                     s))))
+                               (when (string-match fuz-query cand)
+                                 (setq cand (copy-sequence cand))
```
[the remark this answers is not in the document]
Comment (kiennq) [reply lsp-mode.el:4524]: I modified the cached candidates to adding match-data and stuff so just copy the candidates to make sure.

**Pull request merged**